### ISSUE TITLE
Add a Power Transform preprocessor to Shogun.

### ISSUE THREAD

**Issue by Hephaestus12** [shogun-toolbox/shogun#5052]
Like the [PowerTransformer](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.PowerTransformer.html) in Scikit-learn or [boxcox](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.boxcox.html) and [yeojohnson](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.yeojohnson.html) in scipy.stats.

**Comment by karlnapf** [comment]
This shouldn't be too hard to add right?
You could have a go at this... (but let's first finish the poisson regression of course)

**Comment by Hephaestus12** [comment]
Yes, I plan to start working on this after getting done with GLM. :)

**Comment by siddhantjain07** [comment]
@karlnapf may I take this up?

**Comment by gf712** [comment]
@siddhantjain07 Go for it :)

**Comment by siddhantjain07** [comment]
Yes thanks, @gf712!
Could you please tell me, whether it should be
1. A class with methods
2. or a function within some preprocessing code.
and should the file be `.cpp` or a `.hpp`
Sorry if the doubts seem silly, but this my first time fixing some issue at something this big. Kindly bear with me :D

**Comment by gf712** [comment]
This should be a class. No problem, have a look at other classes that inherit from Transform in https://github.com/shogun-toolbox/shogun/tree/develop/src/shogun/preprocessor, it should help get you started :)
